lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -45,6 +45,13 @@
     description:
       - Description of the security group. Required when C(state) is C(present).
     required: false
+  diff:
+    description:
+      -  Add diff mode support with and without check mode. This feature is preview and may change when a common framework is adopted for AWS modules.
+    required: false
+    default: 'False'
+    choices: [ "True", "False" ]
+    version_added: "2.7"
```

Comment:
diff is a task level keyword, not a module option?

Reply:
It is the same level as name, description, vpc_id, etc...

```
ec2_group:
    name: name
    description: description
    vpc_id: vpc-####
    region: us-east-1
    diff: true
```

Reply:
This is the current example listed in the documentation and seems to work.

```json
- name: example2 ec2 group
  ec2_group:
    name: example2
    description: an example2 EC2 group
    vpc_id: 12345
    region: eu-west-1
    rules:
      # 'ports' rule keyword was introduced in version 2.4. It accepts a single port value or a list of values including ranges (from_port-to_port).
      - proto: tcp
        ports: 22
        group_name: example-vpn
      - proto: tcp
        ports:
          - 80
          - 443
          - 8080-8099
        cidr_ip: 0.0.0.0/0
      # Rule sources list support was added in version 2.4. This allows to define multiple sources per source type as well as multiple source types per rule.
      - proto: tcp
        ports:
          - 6379
          - 26379
        group_name:
          - example-vpn
          - example-redis
      - proto: tcp
        ports: 5665
        group_name: example-vpn
        cidr_ip:
          - 172.16.1.0/24
          - 172.16.17.0/24
        cidr_ipv6:
          - 2607:F8B0::/32
          - 64:ff9b::/96
        group_id:
          - sg-edcd9784
  diff: True```

Reply:
Where would I add the documentation as being a task level keyword instead of a module option, where I believe I have it.

Thanks!

Reply:
Task level parameters are [documented here](https://docs.ansible.com/ansible/latest/reference_appendices/playbooks_keywords.html#task). They apply to all tasks, and not present in the documentation for module parameters.